shoop_tests/themes/classic_gray/test_plugins.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import pytest
+from bs4 import BeautifulSoup
+
+from shoop.core.models import Product, ProductCrossSell, StockBehavior
+from shoop.testing.factories import (
+    create_product, get_default_shop, get_default_supplier
+)
+from shoop.themes.classic_gray.plugins import ProductCrossSellsPlugin
+from shoop_tests.front.fixtures import get_jinja_context
+
+
+def _get_class_count(html, html_class):
+    parsed = BeautifulSoup(html)
+    return len(parsed.find_all(class_=html_class))
+
+def _create_cross_sell_products(product, shop, type, product_count):
+    for count in range(product_count):
+        related_product = create_product(
+            "{}-test-sku-{}".format(type, count),
+            shop=shop,
+            supplier=get_default_supplier(),
+            stock_behavior=StockBehavior.UNSTOCKED
+        )
+        ProductCrossSell.objects.create(product1=product, product2=related_product, type=type)
+
+
+@pytest.mark.django_db
+def test_cross_sell_plugin_type():
+    """
+    Test that plugin returns correct number of cross sells when shop contains multiple
+    relation types
+    """
+    shop = get_default_shop()
+    product = create_product("test-sku", shop=shop, stock_behavior=StockBehavior.UNSTOCKED)
+    context = get_jinja_context(product=product)
+    type_counts = (("related", 1),
+                   ("recommended", 2),
+                   ("computed", 3))
+
+    # Create cross sell products and relations in different quantities
+    for type, count in type_counts:
+        _create_cross_sell_products(product, shop, type, count)
+        assert ProductCrossSell.objects.filter(product1=product, type=type).count() == count
+
+    # Make sure quantites returned by plugin match
+    for type, count in type_counts:
+        rendered_plugin = ProductCrossSellsPlugin({"type": type}).render(context)
+        assert _get_class_count(rendered_plugin, "product-box") == count
+
+@pytest.mark.django_db
+def test_cross_sell_plugin_count():
+    shop = get_default_shop()
+    product = create_product("test-sku", shop=shop, stock_behavior=StockBehavior.UNSTOCKED)
+    context = get_jinja_context(product=product)
+
+    _create_cross_sell_products(product, shop, "related", 5)
+    assert ProductCrossSell.objects.filter(product1=product, type="related").count() == 5
+
+    rendered_plugin = ProductCrossSellsPlugin({"type": "related", "count": 3}).render(context)
+    assert _get_class_count(rendered_plugin, "product-box") == 3
```

Comment:
Instead of counting classes in rendered plugin you should probably just test ``shoop.product.get_product_cross_sells`` to return correct amount of products.